core/server/common/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -58,19 +58,39 @@ public SecureHdfsValidationTask(String process) {
 
   @Override
   public boolean validate(Map<String, String> optionsMap) {
-    if (super.shouldSkip()) {
+    if (shouldSkip()) {
       return true;
     }
     if (!super.validate(optionsMap)) {
       return false;
     }
     if (!validatePrincipalLogin()) {
-      System.err.format("Principal login test failed.");
+      System.err.format("Principal login test failed.%n");
       return false;
     }
     return true;
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public boolean shouldSkip() {
+    if (super.shouldSkip()) {
+      return true;
+    }
+    String principal = null;
+    if (Configuration.containsKey(PRINCIPAL_MAP.get(mProcess))) {
+      principal = Configuration.get(PRINCIPAL_MAP.get(mProcess));
+    }
+    if (principal == null || principal.isEmpty()) {
+      System.err.format("Skip validation for secure HDFS. %s is not specified.%n",
+          PRINCIPAL_MAP.get(mProcess).getName());
```

Comment:
could we save `PRINCIPAL_MAP.get(mProcess)` to a varaible?

Reply:
Done.